src/crimson/common/fatal_signal.cc

```diff
@@ -46,7 +46,7 @@ void FatalSignal::install_oneshot_signal_handler()
     sa.sa_flags |= SA_ONSTACK;
   }
   [[maybe_unused]] auto r = ::sigaction(SigNum, &sa, nullptr);
-  assert(r);
+  assert(!r);
```

Comment:
```suggestion
  assert(r == 0);
```
for better readability. IIRC, some statical analyzer (clang-tidy probably) is annoyed if we use a number as a boolean expression.